extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1321,6 +1370,72 @@ export class Project implements ISqlProject {
 		}
 	}
 
+	/**
+	 * Adds a value to project property, separated by semicolon if
+	 * one or more values are already present in project property
+	 * @param propertyName Name of property
+	 * @param valueToAdd Value to be added to project property
+	 */
+	private async addValueToProjectProperty(propertyName: string, valueToAdd: string) {
+		let propertyValue = this.evaluateProjectPropertyValue(propertyName);
+
+		if (propertyValue === undefined) {
+			propertyValue = valueToAdd;
+		} else {
+			if (propertyValue.length > valueToAdd.length
+				&& (propertyValue.startsWith(`${valueToAdd};`)
+					|| propertyValue.endsWith(`;${valueToAdd}`)
+					|| propertyValue.indexOf(`;${valueToAdd};`) > 0)) {
```

Comment:
I'm not sure about these conditions. What if `propertyValue` is `;mysource;` and I'm trying to add `mysource`? It should be no-op right? here it will result in `';mysource;'.indexOf(';mysource;')`, which per V8 is `0`:
![image](https://user-images.githubusercontent.com/11728682/144520226-76610698-f595-4af1-97eb-fd85961b186a.png)
So it will fall-through and add another one? Can you please make sure all the corner cases are covered within add and remove methods?